<a href="https://colab.research.google.com/github/Dania-Yasir/flyrak-project/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the Refresh / Content Opportunity Scoring lane. This lane focuses on helping content reviewers decide which pages should be reviewed first when time and resources are limited.
 I am interested in exploring whether observable search, content, freshness, and engagement signals can be used to create a clear and explainable priority list.
 The final result would support human decisions rather than automatically deciding that a page must be changed.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research question
How can observable search, content, freshness, and engagement signals help prioritize which pages should be reviewed first for a possible content refresh?

### Unit of analysis
The unit of analysis is one content page, represented by one pseudonymized `content_id` and its performance measurements.

### Decision and action
The project will help a content reviewer decide which pages to inspect first when review time is limited. After reviewing a recommended page, the reviewer may choose to refresh, expand, protect, prune, or monitor it.

### Expected output
The expected output is a ranked list of pages with a priority score and clear reasons explaining why each page was recommended.

### Cost of a wrong recommendation
A wrong recommendation may waste editorial time on a page that is not a strong opportunity. It may also cause an important page to be missed or reviewed too late.

### Why data or ML can help
There are many pages and several performance signals to consider at the same time. Data analysis or ML can combine these signals consistently and support the reviewer, but the final decision should still be made by a human.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

# Load the anonymized starter dataset from my GitHub repository
data_url = (
    "https://raw.githubusercontent.com/"
    "Dania-Yasir/flyrak-project/main/"
    "data/raw/content_refresh_anonymized.csv"
)

df = pd.read_csv(data_url)

# 1. Total number of pages
total_pages = len(df)

# 2. Percentage of pages with a declining trend
declining_rate = (
    df["trend_direction"]
    .astype(str)
    .str.lower()
    .eq("down")
    .mean() * 100
)

# 3. Pages that are both stale and still visible
stale_visible_pages = (
    (pd.to_numeric(df["days_since_last_update"], errors="coerce") >= 180)
    & (pd.to_numeric(df["impressions_90d"], errors="coerce") >= 500)
).sum()

print(f"Total pages in the starter dataset: {total_pages:,}")
print(f"Pages with a declining trend: {declining_rate:.1f}%")
print(f"Stale and visible pages: {stale_visible_pages:,}")

Total pages in the starter dataset: 30,000
Pages with a declining trend: 54.2%
Stale and visible pages: 17


### What these numbers suggest

The starter dataset contains 30,000 content pages, which is too many for a reviewer to inspect manually without prioritization. About 54.2% of the pages have a declining trend in the current data window, showing that many pages could potentially require attention. However, only 17 pages meet the simple `stale and visible` rule used here: at least 180 days since the last update and at least 500 impressions in the last 90 days.

These observed numbers support the need for a ranked review system. A single rule may find some useful candidates, but combining multiple observable signals could help reviewers decide which pages deserve attention first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can claim

I can say that the project uses observed search, content, freshness, and engagement signals to support the prioritization of pages for human review. I can report measured patterns in the anonymized dataset and compare whether a ranking method performs better than a simple baseline under the chosen evaluation method.

### What I cannot claim

I cannot claim that the model proves how Google's ranking algorithm works. I also cannot claim that refreshing a recommended page will definitely increase traffic or cause recovery. The output is decision-support only, and every recommendation still requires human review.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.